model/constants.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/python
+
+#
+#
+#
+# All files and artifacts in this repository are licensed under the
+# provisions of the license provided by the LICENSE file in this repository.
+#
+
+#
+# Constant Types and Mappings
+#
+
+# noinspection PyPackageRequirements
+from enum import Enum
+
+class StringStyle(Enum):
+    NONE = "none"
+    CAMEL_CASE = "camelCase"
+    CAMEL_BACK = "camelBack"
+    CAPITAL_CASE = "capitalcase"
+    CONST_CASE = "constcase"
+    LOWER_CASE = "lowercase"
+    PASCAL_CASE = "pascalcase"
+    SENTENCE_CASE = "sentencecase"
+    SNAKE_CASE = "snakecase"
+    SPINAL_CASE = "spinalcase"
+    TITLE_CASE = "titlecase"
+    TRIM_CASE = "trimcase"
+    UPPER_CASE = "uppercase"
+    ALPHANUM_CASE = "alphanumcase"
+
+
+class Unit(Enum):
```

Comment:
Direct string matching is basically what we head before. I added these enums, to have type safe (as far as python gets here) usage of Units, Types and Datatypes. This prevents us from using multiple strings for the same unit (e.g., l, liters, liter) as a check during running vss-tools would fail and allows for type safe generation of code or other artifacts. Right now you'd have to do string matching on "l", "liter", "liters", on a lower-case cased string. In addition, during development you'd always get a list of allowed units, types and datatypes.

Reply:
If it's not clear what I meant, here is an example:
[simple string match](https://gist.github.com/gunnarx/f4e3551e69926855544ef518a4d9b0e7)

It seems to me this does the same thing when it comes to checking VSS files to fulfil a standard ("prevents us from using multiple strings for the same unit (e.g., l, liters, liter)") and is a lot less complex.  

No need to refer to _what we had before_ because I am agreeing it should not  remain exactly the same, instead discussing a way forward.

> Right now you'd have to do string matching on "l", "liter", "liters", on a lower-case cased string

No, I am not advocating that the VSS file should be able to use different spellings for the same thing.  I agree with this check in principle.

> In addition, during development you'd always get a list of allowed units, types and datatypes

When writing custom code that is referring to specific data types then yes, you will be warned if you misspell the name of the units that you explicitly listed, or get IDE support to write them correctly.  I just found the writing of such code to be a rare occasion and I am not convinced that it is worth all the boiler plate code.  But this comes down to a preference.  Whereas I often lean on the side of more formalism, I am in VSS partial to the simplicity that we have created, and keeping the amount of code to maintain to a minimum.
As you also noted, even calling this "type safe" has its limits...